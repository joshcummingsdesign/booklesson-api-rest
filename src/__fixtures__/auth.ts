import { IJwtPayload } from '../auth/interfaces';
import { Auth } from '../auth/entities';

export const auth: Omit<Auth, 'user'> = {
  userId: 1,
  password: 'Changeme!',
};

export const jwtPayload: IJwtPayload = {
  access_token:
    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6Impvc2hjdW1taW5nc2Rlc2lnbkBnbWFpbC5jb20iLCJzdWIiOjEsImlhdCI6MTU5MTc0ODUyMywiZXhwIjoxNTkxNzQ4NTMzfQ.jgWsQFZOdYX0e3oPzAByc3uwDPhYNtfNsk1a8Akp5uE',
};
