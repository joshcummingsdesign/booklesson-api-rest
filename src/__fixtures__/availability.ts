import { Availability } from '../availability/entities';
import { AddAvailabilityDto } from '../availability/dto';

export const availability: Partial<Availability> = {
  id: 999,
  userId: 999,
  datetime: '2020-06-21T00:00:00.000Z',
  available: true,
};

export const availabilities: Partial<Availability>[] = [
  {
    id: 998,
    userId: 999,
    datetime: '2020-06-20T23:00:00.000Z',
    available: false,
  },
  {
    id: 999,
    userId: 999,
    datetime: '2020-06-21T00:00:00.000Z',
    available: true,
  },
];

export const addAvailabilityDto: AddAvailabilityDto = {
  datetime: '2020-06-11T10:00:00Z',
};
