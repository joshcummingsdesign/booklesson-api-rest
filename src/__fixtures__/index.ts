export * from './auth';
export * from './user';
export * from './role';
export * from './instrument';
export * from './lesson';
export * from './availability';
