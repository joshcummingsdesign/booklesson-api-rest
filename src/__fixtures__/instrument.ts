import { Instrument } from '../instrument/entities';

export const instrument: Partial<Instrument> = {
  id: 996,
  name: 'guitar',
};

export const instruments: Partial<Instrument>[] = [
  {
    id: 996,
    name: 'guitar',
  },
  {
    id: 997,
    name: 'piano',
  },
  {
    id: 998,
    name: 'bass',
  },
  {
    id: 999,
    name: 'drums',
  },
];
