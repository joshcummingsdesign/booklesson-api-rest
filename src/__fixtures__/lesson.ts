import { Lesson } from '../lesson/entities';
import { CreateLessonDto } from '../lesson/dto';

type PartialLesson = Omit<Lesson, 'student' | 'teacher'>;

export const lesson: PartialLesson = {
  id: 1,
  studentId: 1,
  teacherId: 2,
  date: '2020-06-10',
  time: '09:00:00',
};

export const lessons: PartialLesson[] = [lesson];

export const createLessonDto: CreateLessonDto = {
  teacherId: 2,
  date: '2020-06-10',
  time: '09:00:00',
};
