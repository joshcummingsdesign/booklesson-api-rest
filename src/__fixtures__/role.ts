import { Role } from '../role/entities';

export const studentRole: Role = {
  id: 998,
  name: 'student',
} as Role;

export const teacherRole: Role = {
  id: 999,
  name: 'teacher',
} as Role;
