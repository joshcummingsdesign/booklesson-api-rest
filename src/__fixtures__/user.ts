import { User } from '../user/entities';
import { Auth } from '../auth/entities';
import { CreateUserDto, UpdateUserDto } from '../user/dto';
import { auth } from './auth';

type ParitalUser = Omit<
  User,
  'auth' | 'availability' | 'lessonsToTake' | 'lessonsToTeach'
>;

export const student: ParitalUser = {
  id: 998,
  firstName: 'Josh',
  lastName: 'Cummings',
  email: 'joshcummingsdesign@gmail.com',
  instrument: 'Bass',
  role: 'student',
};

export const teacher: ParitalUser = {
  id: 999,
  firstName: 'Victor',
  lastName: 'Wooten',
  email: 'victorwooten@gmail.com',
  instrument: 'Bass',
  role: 'teacher',
};

export const studentFull: Omit<User, 'auth'> = {
  ...student,
  availability: [],
  lessonsToTake: [],
  lessonsToTeach: [],
};

export const studentAuth: User = { ...studentFull, auth: auth as Auth };

export const users: ParitalUser[] = [student, teacher];

export const students: ParitalUser[] = [student];

export const teachers: ParitalUser[] = [teacher];

export const createUserDto: CreateUserDto = {
  firstName: 'Josh',
  lastName: 'Cummings',
  email: 'joshcummingsdesign@gmail.com',
  password: 'Changeme!',
  instrument: 'Bass',
  role: 'student',
};

export const updateUserDto: UpdateUserDto = {
  firstName: 'Josh',
  lastName: 'Cummings',
  email: 'joshcummingsdesign@gmail.com',
  instrument: 'Bass',
};
