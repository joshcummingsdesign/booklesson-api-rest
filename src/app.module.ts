import { Module } from '@nestjs/common';
import { ConfigModule } from '@nestjs/config';
import { TypeOrmModule } from '@nestjs/typeorm';
import { validationSchema } from './config';
import { CryptoModule } from './crypto/crypto.module';
import { AuthModule } from './auth/auth.module';
import { RoleModule } from './role/role.module';
import { InstrumentModule } from './instrument/instrument.module';
import { UserModule } from './user/user.module';
import { LessonModule } from './lesson/lesson.module';
import { AvailabilityModule } from './availability/availability.module';

@Module({
  imports: [
    ConfigModule.forRoot({ isGlobal: true, validationSchema }),
    TypeOrmModule.forRoot(),
    CryptoModule,
    AuthModule,
    RoleModule,
    InstrumentModule,
    UserModule,
    LessonModule,
    AvailabilityModule,
  ],
})
export class AppModule {}
