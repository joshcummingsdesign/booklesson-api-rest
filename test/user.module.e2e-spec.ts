import { INestApplication } from '@nestjs/common';
import { Test, TestingModule } from '@nestjs/testing';
import * as request from 'supertest';
import { AppModule } from '../src/app.module';
import { users } from '../src/__fixtures__';

describe('UserModule (e2e)', () => {
  let app: INestApplication;

  beforeAll(async () => {
    const module: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = module.createNestApplication();

    app.init();
  });

  it('GET /api/v1/users', async () => {
    const res = await request(app.getHttpServer())
      .get('/api/v1/users')
      .expect(200);

    expect(res.body).toMatchSnapshot();
  });

  afterAll(async () => {
    await app.close();
  });
});
