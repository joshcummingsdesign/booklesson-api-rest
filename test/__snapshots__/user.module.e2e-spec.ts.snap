// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModule (e2e) GET /api/v1/users 1`] = `
Array [
  Object {
    "email": "joshcummingsdesign@gmail.com",
    "firstName": "Josh",
    "id": 998,
    "instrument": Object {
      "id": 996,
      "name": "guitar",
    },
    "instrumentId": 996,
    "lastName": "Cummings",
    "role": Object {
      "id": 998,
      "name": "student",
    },
    "roleId": 998,
  },
  Object {
    "email": "kurtrosenwinkel@gmail.com",
    "firstName": "Kurt",
    "id": 999,
    "instrument": Object {
      "id": 996,
      "name": "guitar",
    },
    "instrumentId": 996,
    "lastName": "Rosenwinkel",
    "role": Object {
      "id": 999,
      "name": "teacher",
    },
    "roleId": 999,
  },
]
`;
